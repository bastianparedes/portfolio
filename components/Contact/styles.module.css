.linkContainer {
  width: fit-content;
  height: 100%;
  position: absolute;
  top: 0;
  color: rgb(255, 255, 255);
  display: flex;
}

.linksList {
  position: sticky;
  list-style: none;
  bottom: 0;
  margin: auto 0 0;
  padding-top: 4rem;
}

.verticalLine {
  width: 1px;
  height: 10rem;
  background-color: rgb(255, 255, 255);
  margin: 0 auto;
}

@media (max-device-width: 700px) {
  .linkContainer {
    width: fit-content;
    height: fit-content;
    position: relative;
    display: block;
    margin: 0 auto;
    padding: 1rem;
  }

  .linksList {
    padding: 0;
    margin: 0;
    position: relative;
    display: grid;
    grid-template-columns: repeat(4, 1fr);
    gap: 1rem;
  }

  .verticalLine {
    display: none;
  }
}
