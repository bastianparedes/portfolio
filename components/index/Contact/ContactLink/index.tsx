import React from 'react';

import styles from './styles.module.scss';

interface LinkProps {
  link: string;
  children: React.ReactNode;
}

const ContactLink = ({ link, children }: LinkProps): JSX.Element => {
  return (
    <li className={styles.link}>
      <a
        className={styles.link__a}
        href={link}
        target="_blank"
        rel="noreferrer noopener"
      >
        {children}
      </a>
    </li>
  );
};

export default ContactLink;
