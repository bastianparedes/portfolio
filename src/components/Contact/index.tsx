import React from 'react'
import styles from './styles.module.css'
import { AiOutlineLinkedin } from 'react-icons/ai'
import { FiInstagram } from 'react-icons/fi'
import { VscGithubAlt, VscMail } from 'react-icons/vsc'

interface LinkProps {
  link: string
  children?: React.ReactNode
}

const Link = ({ link, children }: LinkProps): JSX.Element => {
  return (
    <li className={styles.link}>
      <a
        className={styles.link__a}
        href={link}
        target="_blank"
        rel="noreferrer noopener"
      >
        {children}
      </a>
    </li>
  )
}

const Contact = (): JSX.Element => {
  return (
    <div className={styles.linkContainer}>
      <ul className={styles.linksList}>
        <Link link="https://github.com/BastianParedes">
          <VscGithubAlt />
        </Link>
        <Link link="mailto:bastian.p.trabajo@outlook.com">
          <VscMail />
        </Link>
        <Link link="https://www.linkedin.com/in/bastianparedes/">
          <AiOutlineLinkedin />
        </Link>
        <Link link="https://www.instagram.com/bastian.p.95/">
          <FiInstagram />
        </Link>
        <div className={styles.verticalLine}></div>
      </ul>
    </div>
  )
}

export default Contact
