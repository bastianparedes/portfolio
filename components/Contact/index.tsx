import React from 'react'
import styles from './styles.module.scss'
import { AiOutlineLinkedin } from 'react-icons/ai'
import { FiInstagram } from 'react-icons/fi'
import { VscGithubAlt, VscMail } from 'react-icons/vsc'
import constants from '../../config/constants.json'
import ContactLink from './ContactLink'

const Contact = (): JSX.Element => {
  return (
    <div className={styles.linkContainer}>
      <ul className={styles.linksList}>
        <ContactLink link={constants.CONTACT.LINKS.GITHUB}>
          <VscGithubAlt />
        </ContactLink>
        <ContactLink link={constants.CONTACT.LINKS.EMAIL}>
          <VscMail />
        </ContactLink>
        <ContactLink link={constants.CONTACT.LINKS.LINKEDIN}>
          <AiOutlineLinkedin />
        </ContactLink>
        <ContactLink link={constants.CONTACT.LINKS.INSTAGRAM}>
          <FiInstagram />
        </ContactLink>
        <div className={styles.verticalLine}></div>
      </ul>
    </div>
  )
}

export default Contact
