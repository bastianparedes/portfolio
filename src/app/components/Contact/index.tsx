import React from 'react';

import { AiOutlineLinkedin } from 'react-icons/ai';
import { FiInstagram } from 'react-icons/fi';
import { VscGithubAlt, VscMail } from 'react-icons/vsc';

import ContactLink from './ContactLink';
import styles from './styles.module.scss';
import constants from '../../../../config/constants';

const Contact = () => {
  return (
    <div className={styles.linkContainer}>
      <ul className={styles.linksList}>
        <ContactLink link={constants.CONTACT.LINKS.GITHUB}>
          <VscGithubAlt />
        </ContactLink>
        <ContactLink link={constants.CONTACT.LINKS.EMAIL}>
          <VscMail />
        </ContactLink>
        <ContactLink link={constants.CONTACT.LINKS.LINKEDIN}>
          <AiOutlineLinkedin />
        </ContactLink>
        <ContactLink link={constants.CONTACT.LINKS.INSTAGRAM}>
          <FiInstagram />
        </ContactLink>
        <div className={styles.verticalLine}></div>
      </ul>
    </div>
  );
};

export default Contact;
