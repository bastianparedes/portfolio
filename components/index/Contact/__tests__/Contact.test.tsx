import React from 'react';
import Contact from '..';
import { render } from '@testing-library/react';

jest.mock('../ContactLink');

describe('<Contact />', () => {
  it('should render', () => {
    const { container } = render(<Contact />);
    expect(container).toMatchSnapshot();
  });
});
