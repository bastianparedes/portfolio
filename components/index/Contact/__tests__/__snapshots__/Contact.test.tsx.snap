// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Contact /> should render 1`] = `
<div>
  <div
    class="linkContainer"
  >
    <ul
      class="linksList"
    >
      <div
        class="verticalLine"
      />
    </ul>
  </div>
</div>
`;
